src/librustdoc/html/static/js/main.js

```diff
@@ -263,7 +263,9 @@ function hideThemeButtonState() {
                 search_input.placeholder = searchState.input.origPlaceholder;
             });
 
-            search_input.removeAttribute('disabled');
+            if (search_input.value != '') {
+                loadSearch();
```

Comment:
Can you check how many times the `search` function is called when you load a page with `?search=something`? I found a few cases before where the function was called more than once. Just to be sure it won't be triggered by this.

Reply:
I just checked, and found that master currently calls `search` twice when you load with `?search=something`, and this branch also calls it twice. So there is a problem, and we should fix it, but it's not introduced or made worse by this PR.

Reply:
Ah so the bug is bug. Well if it's not from this PR then it's fine!